itest/lnd_open_channel_test.go

```diff
@@ -822,3 +822,53 @@ func testSimpleTaprootChannelActivation(ht *lntest.HarnessTest) {
 	// Our test is done and Alice closes her channel to Bob.
 	ht.CloseChannel(alice, chanPoint)
 }
+
+// testOpenChannelLockedBalance tests that when a funding reservation is
+// made for opening a channel, the balance of the required outputs shows
+// up as locked balance in the WalletBalance response.
+func testOpenChannelLockedBalance(ht *lntest.HarnessTest) {
+	// Before we start the test, we'll ensure both sides are connected so
+	// the funding flow can be properly executed.
+	alice := ht.Alice
+	bob := ht.Bob
+	ht.EnsureConnected(alice, bob)
+
+	// We first make sure Alice has no locked wallet balance.
+	balance := alice.RPC.WalletBalance()
+	require.Equal(ht, int64(0), balance.LockedBalance)
```

Comment:
nit: could use `require.EqualValues()` instead of explicit cast.

Reply:
Used `EqualValues` and `NotEqualValues` instead of casts, and checked that the itest fails correctly before the changes, and passes after the changes.